libs/api/src/services/admin/endpoints.ts

```diff
@@ -574,3 +574,35 @@ export type GetWriteInImageResponse = WriteInImageEntry[] | ErrorsResponse;
  */
 export const GetWriteInImageResponseSchema: z.ZodSchema<GetWriteInImageResponse> =
   z.array(WriteInImageEntrySchema);
+
+/**
+ * Schema for {@link GenerateCardPinRequest}.
+ */
+export const GenerateCardPinRequestSchema = z.never();
+
+/**
+ * @url /admin/pins/generate-card-pin
+ * @method POST
+ */
+export type GenerateCardPinRequest = z.TypeOf<
+  typeof GenerateCardPinRequestSchema
+>;
+
+/**
+ * Schema for {@link GenerateCardPinResponse}.
+ */
+export const GenerateCardPinResponseSchema = z.discriminatedUnion('status', [
```

Comment:
I wasn't aware of [`discriminatedUnion`](https://github.com/colinhacks/zod#discriminated-unions)! Sounds like it solves one of the main gripes I have with zod's error messages in that they so often have a bunch of bogus info about the variants that don't match the discriminator.